### DISCUSSION TITLE
Diagnostics information in from_json()

### DISCUSSION THREAD

**Discussion by nestal** [nlohmann/json#4000] [Q&A]
After enabling JSON_DIAGNOTICS, the path to the JSON node that cause the exception will be included in e.what() of the exception throw. This is super helpful.

My question is, when I implement from_json() for my own types, I would need to throw exception to indicate errors. How can I include the JSON path in the exception object I throw in from_json().

On the other hand, is it possible that the caller of from_json() to catch any exception thrown by it and wrap it inside a derived class of nholmann::exception, which includes the diagnotics information?

Thanks in advance.

**Comment by djobet** [comment]
Hello same kind of question for me : I have a from_json function which used to work ok, except for the following caveat :
I used to "validate" the json I got for illegal values (in case I'm processing an old json which is not compatible anymore with the current binary) by throwing out_of_range exceptions. It started breaking when I updated the lib as the create() method now has a 3rd argument : a BasicJsonContext.
Since then, I've read someplace else user code is not supposed to throw exceptions. 

So if I cannot throw an exception what is the correct way to report a value is invalid ?

**Comment by nlohmann** [comment]
Can you provide an example?
